frontend/javascripts/router.tsx

```diff
@@ -456,7 +491,23 @@ class ReactRouter extends React.Component<Props> {
                 path="/datasets/:datasetNameAndId/edit"
                 requiresAdminOrManagerRole
                 render={({ match }: ContextRouter) => {
-                  const datasetId = getDatasetIdFromReadableURLPart(match.params.datasetNameAndId);
+                  const { datasetId, datasetName } = getDatasetIdOrNameFromReadableURLPart(
+                    match.params.datasetNameAndId,
+                  );
+                  if (datasetName) {
```

Comment:
Kinda yes: There are links like `/datasets/:datasetName/view` out in the wild. Very old links. We even have a special backend route to get the organization id for a dataset base only on its name to resolve such old links. See: `getOrganizationForDataset`. 

The problem I am trying to solve here is that a routes ``/datasets/:datasetName/view` and ` `/datasets/:datasetId/view` have the same "matching string" and thus need to be decided by programmatically. I modified `getDatasetIdOrNameFromReadableURLPart` to make some guesswork on whether the passed argument for the match (`match.params.datasetNameAndId`)  contains an actual datasetId or just a name. In case it is a name I fall back to treating the legacy link and trying to resolve it to the newest schema.

As this is necessary for view links. there might also be the same problem for `.../edit` routes. I think it is very unlikely that there are some edit links with there very old URL schema out there. But in case they are out there, this code here takes care of it. I'd say this is not needed but the new return type of `getDatasetIdOrNameFromReadableURLPart` forces the user to consider legacy links. And I think this is a good thing so it won't be forgotten in the future. Therefore, I'd also keep the handling of legacy `/edit` links, bc it feels more case complete to me. But feel free to argue.

Besides that, handling the very old `/view` links is a must :)

Reply:
Thank you for elaborating. Please mention this in a comment as this kind of knowledge gets easily lost otherwise :)